File: src/test/java/com/lethe2211/kotlinsample/java/POJOTest.java
package com.lethe2211.kotlinsample.java;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class POJOTest {

    @Test
    public void POJOTest() {
        POJO pojo = new POJO();
        assertEquals("POJO{num=-1, str=''}", pojo.toString());

        assertEquals(Integer.valueOf(-1), pojo.getNum());

        POJO pojo2 = new POJO(-1, "");
        assertTrue(pojo.equals(pojo2));
    }

}

File: src/test/kotlin/com/lethe2211/kotlinsample/kotlin/POJOTest.kt
package com.lethe2211.kotlinsample.kotlin

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

internal class POJOTest {

    @Test
    fun `Data class`() {
        val pojo = POJO()
        assertEquals("POJO(num=-1, str=)", pojo.toString())

        assertEquals(-1, pojo.num)

        val pojo2 = POJO(-1, "")
        assertTrue(pojo == pojo2)
    }
}